aviary/subsystems/mass/simple_mass/tail.py

```diff
@@ -0,0 +1,281 @@
+        elif tail_type == 'vertical':
+            span_locations = jnp.linspace(0, aircraft__vertical_tail__span, num_sections)
+
+        # Get x_points and dx for later
+        x_points, dx = self.precompute_airfoil_geometry()
+
+        # Thickness distribution
+        thickness_dist = self.airfoil_thickness(x_points, max_thickness)
+
+        if tail_type == 'horizontal':
+            total_mass, _ = quadgk(
+                lambda x: density * self.airfoil_thickness(x, max_thickness) * (
+                    aircraft__horizontal_tail__root_chord - (aircraft__horizontal_tail__root_chord - tip_chord_tail) * (x / aircraft__horizontal_tail__span)
+                    ) * aircraft__horizontal_tail__span, [0, 1], epsabs=1e-9, epsrel=1e-9
+                    )
+            
+            aircraft__horizontal_tail__mass = total_mass
+
+        elif tail_type == 'vertical':
+            total_mass, _ = quadgk(
+                lambda x: density * self.airfoil_thickness(x, max_thickness) * (
+                    aircraft__vertical_tail__root_chord - (aircraft__vertical_tail__root_chord - tip_chord_tail) * (x / aircraft__vertical_tail__span)
+                    ) * aircraft__vertical_tail__span, [0, 1], epsabs=1e-9, epsrel=1e-9
+                    )
+
+            aircraft__vertical_tail__mass = total_mass
+        
+        return aircraft__horizontal_tail__mass, aircraft__vertical_tail__mass
+    
+    def precompute_airfoil_geometry(self):
+        num_sections = self.options['num_sections']
+        n_points = num_sections
+        x_points = jnp.linspace(0, 1, n_points)
+        dx = 1 / (n_points - 1)
+        return x_points, dx
+    
+    def compute_airfoil_geometry(self, chord, camber, camber_location, thickness_dist, x_points, dx):
+
+        section_area = jnp.trapezoid(thickness_dist, x_points, dx=dx) 
+        section_area *= chord
+
+        centroid_x = jnp.trapezoid(x_points * thickness_dist, x_points, dx=dx)
+        centroid_x = (centroid_x * chord) / section_area
+
+        centroid_z = jnp.trapezoid(self.airfoil_camber_line(x_points, camber, camber_location) * thickness_dist, x_points, dx=dx)
+        centroid_z = (centroid_z * chord) / section_area
+        return section_area, centroid_x, centroid_z
+    
+    def airfoil_thickness(self, x, max_thickness):
+        return 5 * max_thickness * (0.2969 * jnp.sqrt(x) - 0.1260 * x - 0.3516 * x**2 + 0.2843 * x**3 - 0.1015 * x**4)
+    
+    def airfoil_camber_line(self, x, camber, camber_location):
+        camber_location = omj.ks_max(camber_location, 1e-9) # Divide by zero check
+        return jnp.where(
+            x < camber_location, 
+            (camber / camber_location**2) * (2 * camber_location * x - x**2), 
+        (camber / (1 - camber_location)**2) * ((1 - 2 * camber_location) + 2 * camber_location * x - x**2)
+        )
+
+    def extract_airfoil_features(self, x_coords, y_coords):
```

Comment:
Yeah, if we process this during setup, it doesn't even need to be jax, and can probably be a bare function.